TestMemTableReservation is flaky

Hit a CI failure on an unrelated PR: https://travis-ci.org/github/cockroachdb/pebble/jobs/727796893

I confirmed it flakes on master, hitting a failure after ~600 runs:
```
stress go test -v -race -run 'TestMemTableReservation'
```
```
=== RUN   TestMemTableReservation
    TestMemTableReservation: db_test.go:755: expected 262144 reserved, but found 524288
--- FAIL: TestMemTableReservation (0.01s)
```

I just hit this in [this run](https://github.com/cockroachdb/pebble/runs/4143056890?check_suite_focus=true).

Can also reliably reproduce locally (on a Linux machine):

```bash
$ go test . -mod vendor -run TestMemTableReservation -count 1000
--- FAIL: TestMemTableReservation (0.00s)
    db_test.go:828: expected 262144 reserved, but found 524288
FAIL
FAIL    github.com/cockroachdb/pebble   0.345s
FAIL
```

I ran this with `-tags invariants` and it provides a little more info:

```
$ go test . -tags 'invariants' -mod vendor -run TestMemTableReservation -count 1000
--- FAIL: TestMemTableReservation (0.00s)
    db_test.go:828: expected 262144 reserved, but found 524288
0x7f311c04e030: memTable buffer was not freed
FAIL    github.com/cockroachdb/pebble   0.123s
FAIL
```

--- a/db_test.go
+++ b/db_test.go
@@ -794,6 +794,9 @@ func TestMemTableReservation(t *testing.T) {
 	}
 	opts.testingRandomized()
 	opts.EnsureDefaults()
+	// We're going to be looking at and asserting the global memtable reservation
+	// amount below so we don't want to race with any triggered stats collections.
+	opts.private.disableTableStats = true
 
 	// Add a block to the cache. Note that the memtable size is larger than the
 	// cache size, so opening the DB should cause this block to be evicted.

